File: cibcc/python/step15/sect1/cibcc.py
import sys
from tokenize import *
from parser import *
from codegen import *

def main():
	if len(sys.argv) != 2:
		print("引数の個数が正しくありません")
		return

	input = sys.argv[1]
	tokens = tokenize(input)
	parser = Parser(tokens)
	prog = parser.program()

	fn = prog
	while fn:
		offset = 0
		var = fn.locals
		while var:
			offset += 8
			var.offset = offset
			var = var.next

		fn.stack_size = (offset + 15) // 16 * 16
		fn = fn.next

	codegen(prog)

if __name__ == "__main__":
	main()

File: cibcc/c++/step14/sect2/cibcc.cpp
#include "cibcc.h"

int main(int argc, char* argv[]) {
	if (argc != 2) {
		std::cout << "引数の個数が正しくありません\n";
	return 1;
	}

	token = tokenize(argv[1]);
	Program *prog = program();

	int offset = 0;
	for (Var *var = prog->locals; var; var = var->next) {
		offset += 8;
		var->offset = offset;
	}
	prog->stack_size = offset;

	codegen(prog);

	return 0;
}

